geopandas/tests/test_geoseries.py

```diff
@@ -488,10 +488,39 @@ def test_from_series(self):
         assert s.index is g.index
 
     # GH 1216
-    def test_expanddim(self):
+    @pytest.mark.parametrize("name", [None, "geometry", "Points"])
+    @pytest.mark.parametrize("crs", [None, "epsg:4326"])
+    def test_reset_index(self, name, crs):
         s = GeoSeries(
             [MultiPoint([(0, 0), (1, 1)]), MultiPoint([(2, 2), (3, 3), (4, 4)])]
         )
         s = s.explode()
+        s.name = name
+        s.crs = crs
         df = s.reset_index()
         assert type(df) == GeoDataFrame
+        # name None -> geometry, otherwise name preserved
+        assert df.geometry.name == name if name is not None else "geometry"
+        assert df.crs == s.crs
+
+    @pytest.mark.parametrize("name", [None, "geometry", "Points"])
+    @pytest.mark.parametrize("crs", [None, "epsg:4326"])
+    def test_to_frame(self, name, crs):
+        s = GeoSeries(
+            [MultiPoint([(0, 0), (1, 1)]), MultiPoint([(2, 2), (3, 3), (4, 4)])]
+        )
+        s = s.explode()
```

Comment:
I think you can leave out the explode in this case (for the test above, it gives a MultiIndex, I suppose)